batchflow/models/validator/validator.py

```diff
@@ -0,0 +1,271 @@
+                - <classification|segmentation|mask|instance|regression>
+                    <classification_kwarg_0>: <value_0>
+                    <classification_kwarg_1>: <value_1>
+                    ...
+                    evaluate:
+                        - <metric_name_0>:
+                            <metric_kwarg_0>: <value_0>
+                            <metric_kwarg_1>: <value_1>
+                        - <metric_name_1>:
+                            ...
+                        ...
+            custom_metrics: (optional, metrics defined in ModelAPI child-class)
+                - <custom_metric_0>
+                - <custom_metric_1>
+                ...
+        ```
+    name : str
+        name of the current model (from yaml)
+    task : str
+        name of the current task (from yaml)
+    """
+    def __init__(self, config=None, **kwargs):
+        if isinstance(config, str):
+            self.config_path = config
+            with open(config) as file:
+                self.config = yaml.load(file, Loader=yaml.Loader)
+        elif isinstance(config, dict):
+            self.config_path = None
+            self.config = deepcopy(config)
+        elif isinstance(config, Config):
+            self.config_path = None
+            self.config = deepcopy(config.config)
+        elif config is None:
+            self.config = {}
+
+        self.config = {**self.config, **kwargs}
+
+        if 'train' in self.config and 'pretrained' in self.config:
+            raise ValueError("Both 'train' and 'pretrained' was founded.")
+
+        self._pretrained = self.config.get('pretrained', {})
+        if 'pretrained' in self.config and self._pretrained is None:
+            self._pretrained = {}
+
+        self._train_ds = {}
+        if 'train' in self.config:
+            if self.config['train'] is not None:
+                self._train_ds = self.config['train'].pop('dataset', {})
+                if isinstance(self._train_ds, str):
+                    self._train_ds = {'path': self._train_ds}
+            else:
+                self.config['train'] = {}
+
+        self._test_ds = {}
+        if 'test' in self.config:
+            if self.config['test'] is not None:
+                self._test_ds = self.config['test'].pop('dataset', {})
+                if isinstance(self._test_ds, str):
+                    self._test_ds = {'path': self._test_ds}
+                self._metrics = self.config['test'].pop('metrics', {})
+
+                if isinstance(self._metrics, str):
+                    self._metrics = [item.strip() for item in self._metrics.split(',')]
+            else:
+                self.config['test'] = {}
+                self._metrics = []
+
+        self.train_dataset = None
+        self.from_train = None
+        self.test_dataset = None
+        self.targets = None
+        self.predictions = None
+
+        self.metrics = {}
+
+    def train_loader(self, path=None, **kwargs):
```

Comment:
Since other methods use verb-based naming this sounds ambiguous (it loads but the name suggests that it trains).
Let's rename it to `load_train`or even `load_train_dataset`